src/main/src/test/java/org/geoserver/config/util/XStreamPersisterTest.java

```diff
@@ -1580,11 +1584,37 @@ public synchronized String toString(Locale locale) {
                         return "foobar";
                     }
                 };
+
         assertTrue(converter.canConvert(anonymous.getClass()));
+
         // wont' try to convert object though
         assertFalse(converter.canConvert(Object.class));
     }
 
+    @Test
+    public void testInternationalStringConverter() throws IOException, ClassNotFoundException {
+        final String message = "This is an unlocalized message";
+        final SimpleInternationalString toTest = new SimpleInternationalString(message);
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        final ObjectOutputStream objectOut = new ObjectOutputStream(out);
+        objectOut.writeObject(toTest);
+        objectOut.close();
+        XStreamPersister persister = new XStreamPersisterFactory().createXMLPersister();
+        Converter candidate =
+                persister
+                        .getXStream()
+                        .getConverterLookup()
+                        .lookupConverterForType(SimpleInternationalString.class);
+        assertTrue(candidate instanceof XStreamPersister.InternationalStringConverter);
+
+        persister.save(toTest, out);
+        final ByteArrayInputStream in = new ByteArrayInputStream(out.toByteArray());
+        final ObjectInputStream objectIn = new ObjectInputStream(in);
+        final Object object = objectIn.readObject();
+        objectIn.close();
```

Comment:
I would covert the byteArrayOutputStream to a String after the save method is called to assert that it contains the message variable.